django/db/models/fields/related.py

```diff
@@ -1541,13 +1540,16 @@ def _check_relationship_model(self, from_model=None, **kwargs):
             if (
                 self.remote_field.through_fields is None
                 and not isinstance(to_model, str)
-                and isinstance(to_model._meta.pk, CompositePrimaryKey)
+                and (
+                    isinstance(to_model._meta.pk, CompositePrimaryKey)
+                    or isinstance(from_model._meta.pk, CompositePrimaryKey)
+                )
             ):
                 errors.append(
                     checks.Error(
-                        "Field defines a relation to the CompositePrimaryKey of model "
-                        f"{self.remote_field.model._meta.object_name!r} which is not "
-                        "supported.",
+                        "Field defines a relation with the CompositePrimaryKey of "
+                        f"model {self.remote_field.model._meta.object_name!r} which is "
+                        "not supported.",
```

Comment:
If the wording of the check is changing, we will need to update it within `docs/ref/checks.txt` :+1: